awscrt/auth.py

```diff
@@ -140,6 +140,28 @@ def new_static(cls, access_key_id, secret_access_key, session_token=None):
         binding = _awscrt.credentials_provider_new_static(access_key_id, secret_access_key, session_token)
         return cls(binding)
 
+    @classmethod
+    def new_profile(
+            cls,
+            client_bootstrap,
+            profile_name='default',
+            config_file_name='~/.aws/config',
+            credentials_file_name='~/.aws/credentials'):
```

Comment:
For the last 3 args, have them be `=None` by default

On windows "~" isn't a thing, so we can't use it as a default string here.
Down in C, you're allowed to passed zeroed-out byte-cursors to get the default values, so take advantage of that and let C deal with the platform-specific home directory stuff